File: Leetcode 0482 License Key Formatting.py
# 20250804
class Solution:
    def licenseKeyFormatting(self, s: str, k: int) -> str:
        s_r = s.replace("-", "").upper()
        if len(s_r) <= k:
            return s_r

        
        left = len(s_r) % k
        ans = []
        if left != 0:
            ans.append(s_r[:left]+"-")
        count = 0
        for i in s_r[left:]: 
            if count == k:
                ans.append("-")
                count = 0
            ans.append(i)  
            count += 1
        
        return "".join(ans)

File: Leetcode 0482 License Key Formatting.go
// 20250804
func licenseKeyFormatting(s string, k int) string {
    s_r := strings.ReplaceAll(strings.ToUpper(s), "-","")
    if len(s_r) <= k {
        return s_r
    }

    left := len(s_r) % k
    var ans []string
    if left != 0 {
        ans = append(ans, s_r[:left]+"-")
    }
    count := 0
    for _, ch := range s_r[left:] {
        if count == k{
            ans = append(ans, "-")
            count = 0
        }
        ans = append(ans, string(ch))
        count += 1
    }

    return strings.Join(ans, "")
}
